sdk/src/azure/iot/az_iot_pnp_client_property.c

```diff
@@ -409,11 +407,5 @@ AZ_NODISCARD az_result az_iot_pnp_client_property_get_next_component_property(
     }
   }
 
-  *out_property_name_and_value = *ref_json_reader;
-
-  // Skip the property value array (if applicable) and move to next token
-  _az_RETURN_IF_FAILED(az_json_reader_skip_children(ref_json_reader));
-  _az_RETURN_IF_FAILED(az_json_reader_next_token(ref_json_reader));
```

Comment:
Before this change, we only need to call skip_children and next_token for internal json reader. But after this change, why are there three calls in the user application?
```
az_json_reader_next_token()
az_json_reader_skip_children()
az_json_reader_next_token()
```

Reply:
So `az_json_reader_skip_children()` will do the skipping for you if the current token is a property name. It does eliminate one call if we wanted as an optimization.
![image](https://user-images.githubusercontent.com/36710865/98283774-fe5aa780-1f54-11eb-8afc-4803d3d0a5f5.png)